### PR TITLE
[Improve] Flink CONF_DIR improvement

### PR DESCRIPTION
**Pull request by wolfboys**
[Improve] Flink CONF_DIR improvement

### CODE REVIEW

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/KubernetesNativeClientTrait.scala [hunk 1]
```diff
@@ -49,16 +48,9 @@ trait KubernetesNativeClientTrait extends FlinkClientTrait {
...
-          flinkConfig
-            .safeSet(KubernetesConfigOptions.KUBERNETES_POD_TEMPLATE, p._2)
-            .safeSet(KubernetesConfigOptions.JOB_MANAGER_POD_TEMPLATE, p._2)
-            .safeSet(KubernetesConfigOptions.TASK_MANAGER_POD_TEMPLATE, p._2)
-        })
-      }
+    // add flink conf configuration, mainly to set the log4j configuration
+    if (!flinkConfig.contains(DeploymentOptionsInternal.CONF_DIR)) {
+      flinkConfig.safeSet(DeploymentOptionsInternal.CONF_DIR, s"${submitRequest.flinkVersion.flinkHome}/conf")
     }
```
Comment (zhoulii) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/KubernetesNativeClientTrait.scala:52]: From what I understand，KubernetesNativeApplicationClient overwrite the setConfig method, this step won't take effect for k8s application mode submit, maybe we can invoke this method in KubernetesNativeApplicationClient#setConfig ?

Comment (wolfboys) [reply streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/KubernetesNativeClientTrait.scala:52]: > From what I understand，KubernetesNativeApplicationClient overwrite the setConfig method, this step won't take effect for k8s application mode submit, maybe we can invoke this method in KubernetesNativeApplicationClient#setConfig ?

Thanks for your review,  updated

**Comment by wolfboys** [comment]
cc @zhoulii PTAL. thx

**Comment by zhoulii** [comment]
LGTM

**Pull request closed without merging**